plugin.yaml

```diff
@@ -86,7 +86,9 @@ providers:
         description: "Kubernetes API Token"
         renderingOptions:
           groupName: Authentication
-          displayType: PASSWORD
+          selectionAccessor: "STORAGE_PATH"
+          valueConversion: "STORAGE_PATH_AUTOMATIC_READ"
+          storage-path-root: "keys"
```

Comment:
@ehe-pd I think we need to add a new field here instead adding the Storage selector to the existing one. 

```suggestion
      - name: token
        type: String
        title: "Token"
        required: false
        description: "Kubernetes API Token"
        renderingOptions:
          groupName: Authentication
      - name: token_path
        type: String
        title: "Token"
        required: false
        description: "Kubernetes API Token"
        renderingOptions:
          groupName: Authentication
          selectionAccessor: "STORAGE_PATH"
          valueConversion: "STORAGE_PATH_AUTOMATIC_READ"
          storage-path-root: "keys"  
```

This happens if I upgrade the plugin to this version
<img width="974" alt="Screen Shot 2022-01-14 at 16 24 09" src="https://user-images.githubusercontent.com/6034968/149573555-d2fed831-16bb-4671-b414-069a40b2e0f8.png">


<img width="1251" alt="Screen Shot 2022-01-14 at 16 24 01" src="https://user-images.githubusercontent.com/6034968/149573546-d5cabe7f-0293-45de-8644-e1e20f7939e2.png">

Reply:
If it is for the purpose of backward compatibility with legacy configurations, I think you are right. 
My concern is adding a new field could make the implementation inconsistent with others and causes specialized logic to creep into the source code. 

Here is my thought:
    If the risk of breaking the existing resource model config is acceptable, I'd like to keep the implementation and avoid introducing any new configuration. 

 To fix the security breach, we have already removed the logic of exposing the token as `RD_CONFIG_KUBERNETES_API_TOKEN`, any existing job steps that rely on it will fail at runtime. No matter we add another new attribute or not, these steps have to be reconfigured by either using their own token or using the k8s config file.

Reply:
Update: 
As discussed with @ltamaster, we agree to add a new field `token_storage_path` to provide backward compatibility.